MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs

```diff
@@ -153,6 +154,46 @@ public override MvxBasePresentationAttribute CreatePresentationAttribute(Type vi
             return base.CreatePresentationAttribute(viewModelType, viewType);
         }
 
+        public override Task<bool> ChangePresentation(MvxPresentationHint hint)
+        {
+            if (hint is MvxPagePresentationHint pagePresentationHint)
+            {
+                var request = new MvxViewModelRequest(pagePresentationHint.ViewModel);
+                var attribute = GetPresentationAttribute(request);
+
+                if (attribute is MvxViewPagerFragmentPresentationAttribute pagerFragmentAttribute)
+                {
+                    ViewPager viewPager = null;
+
+                    // check for a ViewPager inside a Fragment
+                    if (pagerFragmentAttribute.FragmentHostViewType != null)
+                    {
+                        var fragment = GetFragmentByViewType(pagerFragmentAttribute.FragmentHostViewType);
+                        viewPager = fragment.View.FindViewById<ViewPager>(pagerFragmentAttribute.ViewPagerResourceId);
+                    }
+
+                    // check for a ViewPager inside an Activity
+                    if (pagerFragmentAttribute.ActivityHostViewModelType != null)
+                    {
+                        viewPager = CurrentActivity.FindViewById<ViewPager>(pagerFragmentAttribute.ViewPagerResourceId);
+                    }
+
+                    if (viewPager?.Adapter is MvxCachingFragmentStatePagerAdapter adapter)
```

Comment:
Could we somehow make this work with the lowest common denominator, `MvxCachingFragmentPagerAdapter`?

Reply:
Right now our entire implementation for ViewPager uses `MvxCachingFragmentStatePagerAdapter`. I agree it would be better to change that, but probably out of th scope of this PR